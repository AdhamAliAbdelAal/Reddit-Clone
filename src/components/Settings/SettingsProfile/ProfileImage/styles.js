import { Box, styled } from '@mui/system';

export const ProfilePic = styled('img')(() => ({
  outline: '4px solid white',
  borderRadius: '3px 3px 0px 0px ',
  marginTop: 20,
  objectFit: 'cover',
}));
export const AddPhoto = styled(Box)(() => ({
  backgroundColor: 'white',
  borderRadius: '50%',
  width: 36,
  height: 36,
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  position: 'absolute',
  zIndex: '50',

  cursor: 'pointer',

}));
export const InputPhoto = styled('input')(() => ({
  display: 'none',

}));
