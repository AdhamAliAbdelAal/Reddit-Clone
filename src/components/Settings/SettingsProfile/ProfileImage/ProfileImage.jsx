import { Box } from '@mui/material';
import AddAPhotoOutlinedIcon from '@mui/icons-material/AddAPhotoOutlined';
import { useContext, useState, useEffect } from 'react';
import {
  ContentSubHeader, ContentHeader, Content, Text, SubHeader,
} from '../../styles';
import { ProfilePic, AddPhoto } from './styles';
import { SettingsContext } from '../../../../contexts/SettingsProvider';
import { image } from '../../settingsServer';

/**
 * - ProfileImage
 * - Edit Image and Banner in Seetings Page
 *  @component
 */

function ProfileImage() {
  const {
    prefs, setPrefs,
  } = useContext(SettingsContext);
  const [formData, setFormData] = useState();

  useEffect(() => {
    console.log('ss', formData);
  }, [formData]);
  const onFileChange = async (event, type) => {
    const file = event.target.files[0];
    const reader = new FileReader();
    reader.onload = () => {
      if (reader.readyState === 2) {
        if (type === 'profileBackground') {
          setPrefs((oldPrefs) => ({ ...oldPrefs, profileBackground: reader.result }));
        } else {
          setPrefs((oldPrefs) => ({ ...oldPrefs, profilePicture: reader.result }));
        }
        setFormData({
          type,
          file: reader.result,
        });
      }
    };

    if (file && (file.type.match('image/png') || file.type.match('image/jpg') || file.type.match('image/jpeg'))) {
      reader.readAsDataURL(event.target.files[0]);
      image(reader.result, type);
    }
  };

  return (
    <>
      <SubHeader>
        IMAGES
      </SubHeader>
      <Text>
        <Content>
          <ContentHeader>
            Avatar and banner image
          </ContentHeader>
          <ContentSubHeader>
            Images must be .png or .jpg format
          </ContentSubHeader>
        </Content>
        <Box sx={{
          display: 'flex',
          '@media screen and (max-width: 626px)': {
            flexDirection: 'column',
          },
        }}
        >
          <Box sx={{ display: 'flex', marginRight: '18px' }}>
            <Box sx={{
              position: 'relative',
            }}
            >
              <ProfilePic width="120px" height="120px" src={prefs?.profilePicture} alt="user photo" />

              <AddPhoto
                sx={{
                  border: (theme) => `thin solid ${theme.palette.primary?.main}`,
                  position: 'absolute',
                  right: '5%',
                  top: '65%',
                }}
              >
                <label type="file" htmlFor="profilePicture">
                  <input style={{ display: 'none' }} type="file" name="image-upload" accept=".png, .jpg, .jpeg" id="profilePicture" onChange={(e) => { onFileChange(e, 'profilePicture'); }} />
                  <AddAPhotoOutlinedIcon sx={{ cursor: 'pointer', marginTop: '5px' }} color="primary" fontSize="small" />
                </label>
              </AddPhoto>

            </Box>
          </Box>
          <Box sx={{ display: 'flex' }}>
            <Box sx={{
              position: 'relative',
            }}
            >
              <ProfilePic
                sx={{
                  '@media screen and (max-width: 626px)': {
                    width: '250px',
                  },
                }}
                width="412px"
                height="120px"
                src={prefs?.profileBackground}
                alt="user photo"
              />
              <AddPhoto sx={{
                border: (theme) => `thin solid ${theme.palette.primary?.main}`,
                position: 'absolute',
                right: '5%',
                top: '65%',
              }}
              >
                <label type="file" htmlFor="profileBackground">
                  <input style={{ display: 'none' }} type="file" name="image-upload" accept=".png, .jpg, .jpeg" id="profileBackground" onChange={(e) => { onFileChange(e, 'profileBackground'); }} />
                  <AddAPhotoOutlinedIcon sx={{ cursor: 'pointer', marginTop: '5px' }} color="primary" fontSize="small" />
                </label>
              </AddPhoto>
            </Box>
          </Box>
        </Box>
      </Text>
    </>
  );
}

export default ProfileImage;
