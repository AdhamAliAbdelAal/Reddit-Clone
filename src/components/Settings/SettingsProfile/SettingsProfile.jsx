import { Box } from '@mui/system';
import { Switch } from '@mui/material';
import { useEffect, useState } from 'react';
import {
  AntSwitch, ContentSubHeader, ContentHeader, Content, Button, Header,
  SettingsPageConranier, SubHeader,
} from '../styles';
import ProfileInoformation from './ProfileInoformation/ProfileInoformation';
import ProfileImage from './ProfileImage/ProfileImage';
import useFetch from '../../../hooks/useFetch';
import SettingsProvider from '../../../contexts/SettingsProvider';
import settingsPost from '../server';
/**
 * - SettingsProfile
 * - Edit NSFW and Allow people to follow you in Seetings Page
 *  @property {String} data data from fetch
 *  @property {Object} prefs prefs of user
 *  @component
 */
function SettingsProfile() {
  const api = '/users/me/prefs';
  const [data, dataError] = useFetch(api);
  const [prefs, setPrefs] = useState();

  useEffect(() => {
    // console.log(data?.settings.prefs);
    setPrefs(data?.prefs);
    console.log(dataError);
  }, [data, dataError]);
  return (
    data === null ? (<div> error in fecting</div>)
      : (
        <SettingsPageConranier data-testid="settings-profile">
          <Box sx={{ maxWidth: '688px', flex: '1 1 auto' }}>
            <Header>Customize profile</Header>
            <SubHeader>PROFILE INFORMATION</SubHeader>
            <SettingsProvider prefs={prefs} setPrefs={setPrefs}>
              <ProfileInoformation />
            </SettingsProvider>
            <SettingsProvider prefs={prefs} setPrefs={setPrefs}>
              <ProfileImage />
            </SettingsProvider>
            <SubHeader>PROFILE CATEGORY</SubHeader>
            <Button>
              <Content>
                <ContentHeader>
                  NSFW
                </ContentHeader>
                <ContentSubHeader>
                  This content is NSFW (may contain nudity, pornography, profanity or inappropriate content for those under 18)
                </ContentSubHeader>
              </Content>
              <AntSwitch onClick={() => { setPrefs((oldPrefs) => ({ ...oldPrefs, nsfw: !oldPrefs.nsfw })); settingsPost({ ...prefs, nsfw: !prefs?.nsfw }); }}>
                <Switch checked={prefs?.nsfw || false} />
              </AntSwitch>
            </Button>
            <SubHeader>ADVANCED</SubHeader>
            <Button>
              <Content>
                <ContentHeader>
                  Allow people to follow you
                </ContentHeader>
                <ContentSubHeader>
                  Followers will be notified about posts you make to your profile and see them in their home feed.
                </ContentSubHeader>
              </Content>
              <AntSwitch onClick={() => { setPrefs((oldPrefs) => ({ ...oldPrefs, canbeFollowed: !oldPrefs.canbeFollowed })); settingsPost({ ...prefs, canbeFollowed: !prefs?.canbeFollowed }); }}>
                <Switch checked={prefs?.canbeFollowed || false} />

              </AntSwitch>
            </Button>
          </Box>
        </SettingsPageConranier>
      )
  );
}

export default SettingsProfile;
