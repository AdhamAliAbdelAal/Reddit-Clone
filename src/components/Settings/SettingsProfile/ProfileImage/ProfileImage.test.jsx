import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import ProfileImage from './ProfileImage';
import SettingsProfile from '../SettingsProfile';
// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(
    <Router>
      <SettingsProfile>
        <ProfileImage />
      </SettingsProfile>
    </Router>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
