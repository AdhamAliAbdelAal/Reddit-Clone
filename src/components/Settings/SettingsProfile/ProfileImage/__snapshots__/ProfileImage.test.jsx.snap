// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="css-nvyjah"
>
  <div
    className="MuiBox-root css-dp89qc"
  >
    <h2
      className="css-1hww4d4"
    >
      Customize profile
    </h2>
    <h2
      className="css-1izwtqu"
    >
      PROFILE INFORMATION
    </h2>
    <div
      className="css-1yjvs5a"
    >
      <div
        className="css-m67mx2"
      >
        <div
          className="css-15qfu71"
        >
          Display name (optional)
        </div>
        <div
          className="css-1olqyvz"
        >
          Set a display name. This does not change your username.
        </div>
      </div>
      <input
        className="css-7bhg3f"
        maxLength={30}
        onBlur={[Function]}
        onChange={[Function]}
        value=""
      />
      <div
        className="css-708jms"
      >
        30
         
        Character remaining
      </div>
    </div>
    <div
      className="css-1yjvs5a"
    >
      <div
        className="css-m67mx2"
      >
        <div
          className="css-15qfu71"
        >
          About (optional)
        </div>
        <div
          className="css-1olqyvz"
        >
          A brief description of yourself shown on your profile.
        </div>
      </div>
      <textarea
        className="css-n1ebyx"
        maxLength={200}
        onBlur={[Function]}
        onChange={[Function]}
        value=""
      />
      <div
        className="css-708jms"
      >
        200
         
        Character remaining
      </div>
    </div>
    <div
      className="css-1yjvs5a"
    >
      <div
        className="css-m67mx2"
      >
        <div
          className="css-15qfu71"
        >
          Social links (5 max)
        </div>
        <div
          className="css-1olqyvz"
        >
          People who visit your profile will see your social links.
        </div>
      </div>
      <div
        className="css-1n3cfe9"
      >
        <div
          className="MuiBox-root css-70qvj9"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          Add social link
        </div>
      </div>
    </div>
    <h2
      className="css-1izwtqu"
    >
      IMAGES
    </h2>
    <div
      className="css-1yjvs5a"
    >
      <div
        className="css-m67mx2"
      >
        <div
          className="css-15qfu71"
        >
          Avatar and banner image
        </div>
        <div
          className="css-1olqyvz"
        >
          Images must be .png or .jpg format
        </div>
      </div>
      <div
        className="MuiBox-root css-1iu5tr5"
      >
        <div
          className="MuiBox-root css-1574rcx"
        >
          <div
            className="MuiBox-root css-79elbk"
          >
            <img
              alt="user photo"
              className="css-19xvoon"
              height="120px"
              width="120px"
            />
            <div
              className="MuiBox-root css-1izo4jp"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1bptiuh-MuiSvgIcon-root"
                data-testid="AddAPhotoOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 6h-3.17L16 4h-6v2h5.12l1.83 2H21v12H5v-9H3v9c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zM8 14c0 2.76 2.24 5 5 5s5-2.24 5-5-2.24-5-5-5-5 2.24-5 5zm5-3c1.65 0 3 1.35 3 3s-1.35 3-3 3-3-1.35-3-3 1.35-3 3-3zM5 6h3V4H5V1H3v3H0v2h3v3h2z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="MuiBox-root css-k008qs"
        >
          <div
            className="MuiBox-root css-79elbk"
          >
            <img
              alt="user photo"
              className="css-16vxn4c"
              height="120px"
              width="412px"
            />
            <div
              className="MuiBox-root css-1izo4jp"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeSmall css-1bptiuh-MuiSvgIcon-root"
                data-testid="AddAPhotoOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 6h-3.17L16 4h-6v2h5.12l1.83 2H21v12H5v-9H3v9c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zM8 14c0 2.76 2.24 5 5 5s5-2.24 5-5-2.24-5-5-5-5 2.24-5 5zm5-3c1.65 0 3 1.35 3 3s-1.35 3-3 3-3-1.35-3-3 1.35-3 3-3zM5 6h3V4H5V1H3v3H0v2h3v3h2z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <h2
      className="css-1izwtqu"
    >
      PROFILE CATEGORY
    </h2>
    <div
      className="css-m214ph"
    >
      <div
        className="css-m67mx2"
      >
        <div
          className="css-15qfu71"
        >
          NSFW
        </div>
        <div
          className="css-1olqyvz"
        >
          This content is NSFW (may contain nudity, pornography, profanity or inappropriate content for those under 18)
        </div>
      </div>
      <div
        className="css-1au6i0y"
        onClick={[Function]}
      >
        <span
          className="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
        >
          <span
            className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <input
              checked={false}
              className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
          </span>
          <span
            className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </div>
    </div>
    <h2
      className="css-1izwtqu"
    >
      ADVANCED
    </h2>
    <div
      className="css-m214ph"
    >
      <div
        className="css-m67mx2"
      >
        <div
          className="css-15qfu71"
        >
          Allow people to follow you
        </div>
        <div
          className="css-1olqyvz"
        >
          Followers will be notified about posts you make to your profile and see them in their home feed.
        </div>
      </div>
      <div
        className="css-1au6i0y"
        onClick={[Function]}
      >
        <span
          className="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
        >
          <span
            className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <input
              checked={false}
              className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
          </span>
          <span
            className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
